rust/crd/src/authentication.rs

```diff
@@ -0,0 +1,118 @@
+use crate::ObjectRef;
+
+use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
+use stackable_operator::commons::authentication::AuthenticationClassProvider;
+use stackable_operator::{
+    client::Client,
+    commons::authentication::AuthenticationClass,
+    schemars::{self, JsonSchema},
+};
+
+const SUPPORTED_AUTHENTICATION_CLASS: [&str; 1] = ["TLS"];
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("failed to retrieve AuthenticationClass [{}]", authentication_class))]
+    AuthenticationClassRetrieval {
+        source: stackable_operator::error::Error,
+        authentication_class: ObjectRef<AuthenticationClass>,
+    },
+    // TODO: Adapt message if multiple authentication classes are supported
+    #[snafu(display("only one authentication class is currently supported. Possible Authentication classes are {SUPPORTED_AUTHENTICATION_CLASS:?}"))]
+    MultipleAuthenticationClassesProvided,
+    #[snafu(display(
+        "failed to use authentication method [{method}] for authentication class [{authentication_class}] - supported mechanisms: {SUPPORTED_AUTHENTICATION_CLASS:?}",
+    ))]
+    AuthenticationMethodNotSupported {
```

Comment:
I think i would tend towards "provider" vs "method" to match the naming of the AuthClass but minor details

Reply:
changed to provider consistently